src/Item/Item.service.spec.ts

```diff
@@ -123,6 +126,123 @@ describe('Item Service', () => {
         ).rejects.toThrowError(ThirdPartyItemInsertByURNError)
       })
     })
+
+    describe('and the item being upserted contains tags', () => {
+      beforeEach(() => {
+        CollectionService.prototype.getDBCollection = jest.fn()
+      })
+      describe('and it is an insert operation', () => {
+        beforeEach(() => {
+          ;(Item.findByURNSuffix as jest.Mock).mockResolvedValueOnce(undefined)
+        })
+        describe('and it is inserting less than the maximun amount of tags', () => {
+          beforeEach(() => {
+            dbItem = {
+              ...dbItemMock,
+              eth_address: '0xAddress',
+              data: {
+                ...dbItemMock.data,
+                tags: Array(MAX_TAGS_LENGTH - 1).fill('tag'),
+              },
+            }
+            dbCollectionMock.eth_address = dbItem.eth_address
+            mockOwnableCanUpsert(Item, dbItem.id, dbItem.eth_address, true)
+            mockOwnableCanUpsert(
+              Collection,
+              dbCollectionMock.id,
+              dbItem.eth_address,
+              true
+            )
+            ;(CollectionService.prototype
+              .getDBCollection as jest.Mock).mockResolvedValueOnce(
+              dbCollectionMock
+            )
+            ;(Item.upsert as jest.Mock).mockResolvedValueOnce(dbItem)
+            CollectionService.prototype.isDCLPublished = jest.fn()
+          })
+          it('should not throw any errors and return the inserted item', () => {
+            const result = service.upsertItem(
+              Bridge.toFullItem(dbItem, dbTPCollectionMock),
+              dbItem.eth_address
+            )
+            return expect(result).resolves.toEqual(
+              Bridge.toFullItem(dbItem, dbCollectionMock)
+            )
+          })
+        })
+
+        describe('and it is inserting more than the maximun amount of tags', () => {
+          beforeEach(() => {
+            dbItem = {
+              ...dbItemMock,
+              data: {
+                ...dbItemMock.data,
+                tags: Array(MAX_TAGS_LENGTH + 1).fill('tag'),
+              },
+            }
+          })
+          it('should throw the MaximunAmountOfTagsReachedError error', () => {
+            return expect(() =>
+              service.upsertItem(
+                Bridge.toFullItem(dbItem, dbTPCollectionMock),
+                dbItem.eth_address
+              )
+            ).rejects.toThrowError(MaximunAmountOfTagsReachedError)
+          })
+        })
+      })
+
+      describe('and it is an update operation', () => {
+        beforeEach(() => {
+          ;(Item.findByURNSuffix as jest.Mock).mockResolvedValueOnce(dbItem)
+        })
+        describe('and the item already has the maximun amount of tags', () => {
+          it('should throw the MaximunAmountOfTagsReachedError error', () => {
+            return expect(() =>
```

Comment:
The rejects modifier expects a promise, so It shouldn't be necessary to wrap the expectation on a function, would you mind removing it?
```suggestion
            return expect(
```